hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java

```diff
@@ -113,5 +114,10 @@ public static FlinkWriteHelper newInstance() {
       hoodieRecord.setCurrentLocation(rec1.getCurrentLocation());
       return hoodieRecord;
     }).orElse(null)).filter(Objects::nonNull).collect(Collectors.toList());
+
+    if (hasInsert) {
+      recordList.get(0).getCurrentLocation().setInstantTime("I");
+    }
+    return recordList;
```

Comment:
In line 114, we already reset the location, so each records list under the same key after reduction should have the same instant time type as before, so why the set is needed ?

Reply:
I wrote a test in local and found out the order of the list was changed after reduction. <id1, id2> became <id2,id1> somehow, so it's not related to a single record.

Reply:
Yes, the Map::values does not guarantee the sequence, state index based writer has no problem because it assigns the instant "I" and "U" based on the buckets of last checkpoint, and reuse the buckets within one checkpoint.

This fix is necessary for it to be more robust.